renku/core/commands/workflow.py

```diff
@@ -301,3 +302,73 @@ def _export_workflow(
 def export_workflow_command():
     """Command that exports a workflow into a given format."""
     return Command().command(_export_workflow).require_clean().with_database(write=False)
+
+
+@inject.autoparams()
+def _lookup_paths_in_paths(client_dispatcher: IClientDispatcher, lookup_paths: List[str], target_paths: List[str]):
+    """Return all lookup_paths that are in or under target_paths."""
+    client = client_dispatcher.current_client
+
+    dirs = []
+    files = set()
+
+    for p in lookup_paths:
+        path = Path(p).resolve().relative_to(client.path)
+        if path.is_dir():
+            dirs.append(str(path))
+        else:
+            files.add(path)
+
+    target_dirs = []
+    target_files = set()
+
+    for p in target_paths:
+        path = Path(p).resolve().relative_to(client.path)
+        if path.is_dir():
+            target_dirs.append(str(path))
+        else:
+            target_files.add(path)
+
+    result = set()
+
+    for target_file in target_files:
+        if target_file in files or any(str(target_file).startswith(d) for d in dirs):
```

Comment:
It's safer to check `any(Path(d) in Path(target_file).parents for d in dirs)` since `startswith` can be true for partial matches.